### PR TITLE
Feature/39712 GIAS API integration

### PR DESCRIPTION
**Pull request by GuyHarwood**
Supporting code to communicate with GIAS SOAP Api.

Partially complete feature - this PR deals with downloading, parsing and storing the extract file.

Further work will process the extract data.

### CODE REVIEW

File: tslib/src/functions/gias-sync/gias.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+      fail('error was expected to be thrown')
+    } catch (error) {
+      expect(error).toBeDefined()
+      expect(error.message).toEqual('gias password is required')
+    }
+  })
+
+  test.todo('GetExtract: verify returned data structure')
+
+  test.skip('GetExtract:should return an empty object if no results', async () => {
```
Comment (jon-shipley) [inline tslib/src/functions/gias-sync/gias.service.spec.ts:185]: Why are there 3 skipped tests?

File: tslib/src/functions/gias-sync/index.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  const start = performance.now()
+
+  const giasService = new GiasService()
+  const extractXml = await giasService.GetExtract(config.Gias.ExtractId)
+  context.bindings.giasExtractFile = extractXml
+
+  const end = performance.now()
+  const durationInMilliseconds = end - start
+  const timeStamp = new Date().toISOString()
+  context.log(`${functionName}: ${timeStamp} run complete: ${durationInMilliseconds} ms`)
```
Comment (jon-shipley) [inline tslib/src/functions/gias-sync/index.ts:17]: How fast do you expect this to run? ms / s / mm ?

**Review by jon-shipley — APPROVED** [review]
LGTM 👍 

Approving on the basis that there is nothing to hold this up, and the skipped tests will be addressed in a future PR

File: tslib/src/functions/gias-sync/index.ts (hunk 2 above)
Comment (GuyHarwood) [reply tslib/src/functions/gias-sync/index.ts:17]: it usually takes between 3000 - 6000ms

The function is incomplete, as it only persists the extract.   Processing is yet to be implemented.  I will summarise this in the PR description.

File: tslib/src/functions/gias-sync/gias.service.spec.ts (hunk 1 above)
Comment (GuyHarwood) [reply tslib/src/functions/gias-sync/gias.service.spec.ts:187]: they are integration tests that are useful for debugging the real payload.  have added comments to indicate their purpose.  will be removed or moved to suitable location before feature completion.

**Comment by GuyHarwood** [comment]
@jon-shipley regards the deep path concern that you mentioned a little while back - i have addressed that with a dedicated parser

**Pull request merged**